core/src/main/java/org/apache/carbondata/core/indexstore/blockletindex/BlockIndex.java

```diff
@@ -793,9 +794,16 @@ private boolean addBlockBasedOnMinMaxValue(FilterExecuter filterExecuter, byte[]
     BitSet bitSet = null;
     if (filterExecuter instanceof ImplicitColumnFilterExecutor) {
       String uniqueBlockPath;
-      if (segmentPropertiesWrapper.getCarbonTable().isHivePartitionTable()) {
-        uniqueBlockPath = filePath
-            .substring(segmentPropertiesWrapper.getCarbonTable().getTablePath().length() + 1);
+      CarbonTable carbonTable = segmentPropertiesWrapper.getCarbonTable();
+      if (carbonTable.isHivePartitionTable()) {
+        boolean isStandardTable = CarbonUtil.isStandardCarbonTable(carbonTable);
+        // While data loading to SI created on Partition table, on partition directory, '/' will be
+        // replaced with '#', to support multi level partitioning. For example, BlockId will be
+        // look like `part1=1#part2=2/xxxxxxxxx`. During query also, blockId should be
+        // replaced by '#' in place of '/', to match and prune data on SI table.
+        uniqueBlockPath = CarbonUtil.getBlockId(carbonTable.getAbsoluteTableIdentifier(), filePath,
+                "", carbonTable.isTransactionalTable(),
+                isStandardTable, carbonTable.isHivePartitionTable());
```

Comment:
in this case, the following variable or method are constant, right?
isStandardTable is false, 
carbonTable.isHivePartitionTable() is true

Reply:
Handled